static/js/tools/map/app.tsx

```diff
@@ -114,9 +115,16 @@ function updateHash(context: ContextType): void {
   let hash = updateHashStatVar("", context.statVar.value);
   hash = updateHashPlaceInfo(hash, context.placeInfo.value);
   hash = updateHashDisplay(hash, context.display.value);
+  // leaflet flag is part of the search arguments instead of hash, so need to
+  // update that separately
+  let args = "";
+  if (context.display.value.allowLeaflet) {
+    args += `?${ALLOW_LEAFLET_URL_ARG}=1`;
+  }
   const newHash = encodeURIComponent(hash);
   const currentHash = location.hash.replace("#", "");
-  if (newHash && newHash !== currentHash) {
-    history.pushState({}, "", `${MAP_REDIRECT_PREFIX}#${newHash}`);
+  const currentArgs = location.search;
+  if (newHash && (newHash !== currentHash || args !== currentArgs)) {
+    history.pushState({}, "", `${MAP_REDIRECT_PREFIX}${args}#${newHash}`);
```

Comment:
We should forward along all args (not just leaflet), and then append the hash. Query args are a good place for persistent effects (e.g. allow various features). This works for now, and we can update that in a TODO.

Perhaps also rename `MAP_REDIRECT_PREFIX` to `MAP_URL_PATH` so we don't accidentally add other args / path to the const.

Reply:
added todo & renamed the constant